src/ngraph/pass/reshape_sinking.cpp

```diff
@@ -247,32 +278,37 @@ static void materialize_shapes(shared_ptr<Node> n,
         auto arg = n->get_argument(i);
         if (reorders.count(arg) != 0)
         {
-            NGRAPH_DEBUG << "Materializing " << describe_reshape(reorders.at(arg)) << " for "
+            auto arg_reshape = reorders.at(arg);
+            NGRAPH_DEBUG << "Materializing " << describe_reshape(arg_reshape) << " for "
                          << arg->get_name();
-            mark_reshape_for_deletion(reorders.at(arg), reshapes_to_delete);
-            if (reorders.at(arg)->get_input_order() != get_default_order(arg->get_shape()))
+            mark_reshape_for_deletion(arg_reshape, reshapes_to_delete);
+            auto arg_shape = arg->get_shape();
+            if (arg_reshape->get_input_order() != get_default_order(arg->get_shape()))
             {
                 // Insert if arg needs to be transposed.
-                insert_reshape(n, reorders.at(arg), i);
+                insert_reshape(n, arg_reshape, i);
             }
             //no swimming up
         }
     }
-    reorders[n] = create_default_reshape(n);
+    write_reshapemap(reorders, n, create_default_reshape(n));
 }
 
 static void sink_reshape(shared_ptr<op::Reshape> reshape,
                          ReshapeMap& reorders,
                          set<shared_ptr<Node>>& reshapes_to_delete)
 {
+    NGRAPH_DEBUG << "Sinking Reshape :" << describe_reshape(reshape);
     auto orig_reshape = reorders.at(reshape->get_argument(0));
-    if (!reshape->get_is_transpose())
+    // 1) Not a Transpose or 2) Rank changing operation.
```

Comment:
This PR has other other line changes but they are just there for helping debug printing. This IF condition change is the only functional change.